openassessment/xblock/static/js/src/lms/oa_grade.js

```diff
@@ -20,18 +22,24 @@ OpenAssessment.GradeView.prototype = {
         var view = this;
         var baseView = this.baseView;
         var stepID = ".step--grade";
+        var focusID = "[id='oa_grade_" + usageID + "']";
+        view.isRendering = true;
         this.server.render('grade').done(
             function(html) {
                 // Load the HTML and install event handlers
                 $(stepID, view.element).replaceWith(html);
                 view.server.renderLatex($(stepID, view.element));
+                view.isRendering = false;
                 view.installHandlers();
-                if (typeof usageID !== 'undefined' &&
-                    !$(stepID, view.element).hasClass("is--unfinished") &&
-                    !$(stepID, view.element).hasClass("is--unstarted") &&
-                    !$(stepID, view.element).hasClass("is--waiting--staff")) {
-                    $("[id='oa_grade_" + usageID + "']", view.element).focus();
-                }
+
+                view.baseView.announceStatusChangeToSRandFocus(view.announceStatus,
+                    stepID,
```

Comment:
Are you comfortable with announceStatusChangeToSRandFocus checking view.announceStatus directly and setting it to false at the end of the method (since view is already passed into the method)?